### ISSUE TITLE
With angular-gridster, endless digest loop on item resize

### ISSUE THREAD

**Issue by pihomeserver** [n3-charts/line-chart#378]
Hi,

Using linechart directive in a directive used as a widget in a dashboard built using gridster, I have two issues : the chart does not take the full width/heigh but i will handle that later as for now when i resize a gridster element i have a crash saying that the app went in a digest endless loop.
Here is a widget content (template of a directive):
```
<div class="widget-main">
	<div ng-show="widget.loadingData" class="loader">Loading...</div>
	<widgetContentElement ng-show="!widget.loadingData"></widgetContentElement>
</div>
```
widgetContentElement directive is 
```
<div id="linechartParent">
	<linechart data="chartData" options="chartOptions" mode="" width="" height=""></linechart>
</div>
```
chartData is a new empty array and chartOptions is 
```
			$scope.chartOptions = {
			  axes: {
			    x: {key: 'x', type: 'date'},
			    y: {type: 'linear', ticks: 5, innerTicks: true, grid: true}
			  },
			  series: [
			    {y: 'value', color: 'steelblue', thickness: '1px', type: 'area', striped: true, label: 'Pouet'}
			  ],
			  lineMode: 'linear',
			  tension: 0.7,
			  tooltip: {mode: 'scrubber', formatter: function(x, y, series) {return 'pouet';}},
			  drawLegend: false,
			  drawDots: false,
			  hideOverflow: false,
			  columnsHGap: 5
			}
```
I use the latest version (master branch not the v2) and i checks that the PR #239 is included

The error message is the following and i see that it seems to come from width and/or height that are always updating
```
10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: [[{"msg":"options","newVal":{"axes":{"x":{"key":"x","type":"date"},"y":{"type":"linear","ticks":5,"innerTicks":true,"grid":true}},"series":[{"y":"value","color":"steelblue","thickness":"1px","type":"area","striped":true,"label":"Pouet","axis":"y","dotSize":2,"id":"series_0"}],"lineMode":"linear","tension":0.7,"tooltip":{"mode":"scrubber"},"drawLegend":false,"drawDots":false,"hideOverflow":false,"columnsHGap":5,"stacks":[],"margin":{"top":20,"right":50,"bottom":60,"left":50,"width":703,"height":775},"toggle":true},"oldVal":{"axes":{"x":{"key":"x","type":"date"},"y":{"type":"linear","ticks":5,"innerTicks":true,"grid":true}},"series":[{"y":"value","color":"steelblue","thickness":"1px","type":"area","striped":true,"label":"Pouet","axis":"y","dotSize":2,"id":"series_0"}],"lineMode":"linear","tension":0.7,"tooltip":{"mode":"scrubber"},"drawLegend":false,"drawDots":false,"hideOverflow":false,"columnsHGap":5,"stacks":[],"margin":{"top":20,"right":50,"bottom":60,"left":50,"width":703,"height":625},"toggle":true}}],[{"msg":"options","newVal":"...","oldVal":{"axes":{"x":{"key":"x","type":"date"},"y":{"type":"linear","ticks":5,"innerTicks":true,"grid":true}},"series":[{"y":"value","color":"steelblue","thickness":"1px","type":"area","striped":true,"label":"Pouet","axis":"y","dotSize":2,"id":"series_0"}],"lineMode":"linear","tension":0.7,"tooltip":{"mode":"scrubber"},"drawLegend":false,"drawDots":false,"hideOverflow":false,"columnsHGap":5,"stacks":[],"margin":{"top":20,"right":50,"bottom":60,"left":50,"width":703,"height":650},"toggle":true}}],[{"msg":"options","newVal":"...","oldVal":{"axes":{"x":{"key":"x","type":"date"},"y":{"type":"linear","ticks":5,"innerTicks":true,"grid":true}},"series":[{"y":"value","color":"steelblue","thickness":"1px","type":"area","striped":true,"label":"Pouet","axis":"y","dotSize":2,"id":"series_0"}],"lineMode":"linear","tension":0.7,"tooltip":{"mode":"scrubber"},"drawLegend":false,"drawDots":false,"hideOverflow":false,"columnsHGap":5,"stacks":[],"margin":{"top":20,"right":50,"bottom":60,"left":50,"width":703,"height":675},"toggle":true}}],[{"msg":"options","newVal":"...","oldVal":{"axes":{"x":{"key":"x","type":"date"},"y":{"type":"linear","ticks":5,"innerTicks":true,"grid":true}},"series":[{"y":"value","color":"steelblue","thickness":"1px","type":"area","striped":true,"label":"Pouet","axis":"y","dotSize":2,"id":"series_0"}],"lineMode":"linear","tension":0.7,"tooltip":{"mode":"scrubber"},"drawLegend":false,"drawDots":false,"hideOverflow":false,"columnsHGap":5,"stacks":[],"margin":{"top":20,"right":50,"bottom":60,"left":50,"width":703,"height":700},"toggle":true}}],[{"msg":"options","newVal":"...","oldVal":{"axes":{"x":{"key":"x","type":"date"},"y":{"type":"linear","ticks":5,"innerTicks":true,"grid":true}},"series":[{"y":"value","color":"steelblue","thickness":"1px","type":"area","striped":true,"label":"Pouet","axis":"y","dotSize":2,"id":"series_0"}],"lineMode":"linear","tension":0.7,"tooltip":{"mode":"scrubber"},"drawLegend":false,"drawDots":false,"hideOverflow":false,"columnsHGap":5,"stacks":[],"margin":{"top":20,"right":50,"bottom":60,"left":50,"width":703,"height":725},"toggle":true}}]]
```

Any help is welcome !

**Comment by pihomeserver** [comment]
Ok found that it's looping and adding 25px to the element height after each loop. And i don't know why
A way to solve it is to force parent size with following code :
```javascript
document.getElementById('linechartParent').setAttribute("style","width:"+newHeight+"px");
document.getElementById('linechartParent').setAttribute("style","height:"+newHeight+"px");
$scope.chartOptions.foo = Math.random();
```
I force the size of the container and then update options on which there is a watch

**Comment by pihomeserver** [comment]
Bad news : it works on Safari, not on Chrome ....

**Comment by pihomeserver** [comment]
Partially solved by #372 (no more error but resized is triggered only at the end of a css transformation)
